README


This archive contains the distribution Dist-Zilla-PluginBundle-GopherRepellent,
version 1.002:

  keep those pesky gophers out of your dists!

This software is copyright (c) 2010 by Randy Stauner.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.




Changes
Revision history for Dist-Zilla-PluginBundle-GopherRepellent

1.002 2011-04-11

  [Deprecations]
  - The current namespace is deprecated.
    This always was/should have been an Author bundle.

  [Add Dist::Zilla Plugins]
  - Prepender
  - Git::NextVersion (instead of Git::DescribeVersion)

1.001 2011-02-10

  - Manage Changes respectably by replacing GitFmtChanges plugin
  with NextRelease, CheckChangesHasContent, and CPANChangesTests

1.000007 2011-01-19

  - remove Moose::Autobox dependency
  - allow attributes to be overridden in dist.ini for any included plugin
  - add 'skip_plugins' option

0.009007 2010-12-30

  - Use Pod::Weaver::Plugin::WikiDoc

0.008004 2010-12-16

  - Require Dist::Zilla::Stash::PodWeaver

  [Add Dist::Zilla Plugins]
  - Authority
  - DualBuilders (prefer 'build')
  - PodLinkTests if available
  - PodSpellingTests
  - ReportVersions::Tiny

  [Add Pod::Weaver Plugins]
  - StopWords

0.004001 2010-11-17

  - Use Dist::Zilla::Plugin::GithubMeta
  - Lower required version of Pod::Weaver::Section::Support

0.003011 2010-11-15

  - Initial Release to CPAN


Build.PL

use strict;
use warnings;

use Module::Build 0.3601;


my %module_build_args = (
  'build_requires' => {
    'File::Find' => '0',
    'File::Temp' => '0',
    'Module::Build' => '0.3601',
    'Test::More' => '0.96'
  },
  'configure_requires' => {
    'Module::Build' => '0.3601'
  },
  'dist_abstract' => 'keep those pesky gophers out of your dists!',
  'dist_author' => [
    'Randy Stauner <rwstauner@cpan.org>'
  ],
  'dist_name' => 'Dist-Zilla-PluginBundle-GopherRepellent',
  'dist_version' => '1.002',
  'license' => 'perl',
  'module_name' => 'Dist::Zilla::PluginBundle::GopherRepellent',
  'recommends' => {
    'Dist::Zilla::Plugin::PodLinkTests' => '1.003'
  },
  'recursive_test_files' => 1,
  'requires' => {
    'Dist::Zilla' => '4.102345',
    'Dist::Zilla::Plugin::Authority' => '1.001',
    'Dist::Zilla::Plugin::Bugtracker' => '0',
    'Dist::Zilla::Plugin::CPANChangesTests' => '0',
    'Dist::Zilla::Plugin::CheckChangesHasContent' => '0.003',
    'Dist::Zilla::Plugin::CompileTests' => '1.100740',
    'Dist::Zilla::Plugin::DualBuilders' => '1.001',
    'Dist::Zilla::Plugin::Git::NextVersion' => '0',
    'Dist::Zilla::Plugin::GithubMeta' => '0.10',
    'Dist::Zilla::Plugin::KwaliteeTests' => '0',
    'Dist::Zilla::Plugin::MetaNoIndex' => '1.101130',
    'Dist::Zilla::Plugin::MetaProvides::Package' => '1.11044404',
    'Dist::Zilla::Plugin::MinimumPerl' => '0.02',
    'Dist::Zilla::Plugin::MinimumVersionTests' => '0',
    'Dist::Zilla::Plugin::NextRelease' => '0',
    'Dist::Zilla::Plugin::PkgVersion' => '0',
    'Dist::Zilla::Plugin::PodCoverageTests' => '0',
    'Dist::Zilla::Plugin::PodSpellingTests' => '0',
    'Dist::Zilla::Plugin::PodSyntaxTests' => '0',
    'Dist::Zilla::Plugin::PodWeaver' => '0',
    'Dist::Zilla::Plugin::PortabilityTests' => '0',
    'Dist::Zilla::Plugin::Prepender' => '1.100960',
    'Dist::Zilla::Plugin::ReportVersions::Tiny' => '1.01',
    'Dist::Zilla::Plugin::Repository' => '0.16',
    'Dist::Zilla::Plugin::TaskWeaver' => '0.101620',
    'Dist::Zilla::PluginBundle::Basic' => '0',
    'Dist::Zilla::Role::PluginBundle::Easy' => '0',
    'Dist::Zilla::Stash::PodWeaver' => '1.001000',
    'Moose' => '0',
    'Pod::Elemental' => '0.102360',
    'Pod::Elemental::Transformer::List' => '0',
    'Pod::Weaver' => '3.101632',
    'Pod::Weaver::Config::Assembler' => '0',
    'Pod::Weaver::Plugin::StopWords' => '1.001005',
    'Pod::Weaver::Plugin::Transformer' => '0',
    'Pod::Weaver::Plugin::WikiDoc' => '0',
    'Pod::Weaver::PluginBundle::Default' => '0',
    'Pod::Weaver::Section::Support' => '1.001',
    'perl' => '5.006'
  },
  'script_files' => []
);


my $build = Module::Build->new(%module_build_args);

$build->create_build_script;


lib/Pod/Weaver/PluginBundle/GopherRepellent.pm
#
# This file is part of Dist-Zilla-PluginBundle-GopherRepellent
#
# This software is copyright (c) 2010 by Randy Stauner.
#
# This is free software; you can redistribute it and/or modify it under
# the same terms as the Perl 5 programming language system itself.
#
package Pod::Weaver::PluginBundle::GopherRepellent;
BEGIN {
  $Pod::Weaver::PluginBundle::GopherRepellent::VERSION = '1.002';
}
BEGIN {
  $Pod::Weaver::PluginBundle::GopherRepellent::AUTHORITY = 'cpan:RWSTAUNER';
}
# ABSTRACT: keep those pesky gophers out of your POD!

use strict;
use warnings;

use Pod::Weaver 3.101632 ();
use Pod::Weaver::PluginBundle::Default ();
use Pod::Weaver::Plugin::StopWords 1.001005 ();
use Pod::Weaver::Plugin::Transformer ();
use Pod::Weaver::Plugin::WikiDoc ();
use Pod::Weaver::Section::Support 1.001 ();
use Pod::Elemental 0.102360 ();
use Pod::Elemental::Transformer::List ();

use Pod::Weaver::Config::Assembler;
sub _exp { Pod::Weaver::Config::Assembler->expand_package($_[0]) }

sub _plain {
	my ($plug) = @_;
	(my $name = $plug) =~ s/^\W//;
	return [ $name, _exp($plug), {} ];
}

sub _bundle_name {
	my $class = @_ ? ref $_[0] || $_[0] : __PACKAGE__;
	join('', '@', ($class =~ /([^:]+)$/));
}

sub mvp_bundle_config {
	my ($self, $bundle) = @_;
	my @plugins;

	# NOTE: bundle name gets prepended to each plugin name at the end

	push @plugins, (
		# plugin
		_plain('-WikiDoc'),
		# default
		_plain('@CorePrep'),

		# sections
		# default
		_plain('Name'),
		_plain('Version'),

		[ 'Prelude',     _exp('Region'),  { region_name => 'prelude' } ],
	);

	for my $plugin (
		# default
		[ 'Synopsis',    _exp('Generic'), {} ],
		[ 'Description', _exp('Generic'), {} ],
		[ 'Overview',    _exp('Generic'), {} ],
		# extra
		[ 'Usage',       _exp('Generic'), {} ],

		# default
		[ 'Attributes',  _exp('Collect'), { command => 'attr'   } ],
		[ 'Methods',     _exp('Collect'), { command => 'method' } ],
		[ 'Functions',   _exp('Collect'), { command => 'func'   } ],
	) {
		$plugin->[2]{header} = uc $plugin->[0];
		push @plugins, $plugin;
	}

	# default
	push @plugins, (
		_plain('Leftovers'),
		[ 'Postlude',    _exp('Region'),    { region_name => 'postlude' } ],

		# TODO: consider SeeAlso if it ever allows comments with the links

		# extra
		# include Support section with various cpan links and github repo
		[ 'Support',     _exp('Support'),
			{ repository_content => '', repository_link => 'both' }
		],

		# default
		_plain('Authors'),
		_plain('Legal'),

		# plugins
		[ 'List',        _exp('-Transformer'), { 'transformer' => 'List' } ],
		_plain('-StopWords'),
	);

	# prepend bundle name to each plugin name
	my $name = $self->_bundle_name;
	@plugins = map { $_->[0] = "$name/$_->[0]"; $_ } @plugins;

	return @plugins;
}

1;


__END__
=pod

=for :stopwords Randy Stauner PluginBundle WikiDoc

=head1 NAME

Pod::Weaver::PluginBundle::GopherRepellent - keep those pesky gophers out of your POD!

=head1 VERSION

version 1.002

=head1 DEPRECATED

This module is deprecated.
It will soon be renamed into the Author namespace.

=head1 SYNOPSIS

	# weaver.ini

	[@GopherRepellent]

or with a F<dist.ini> like so:

	# dist.ini

	[@GopherRepellent]

you don't need a F<weaver.ini> at all.

=head1 DESCRIPTION

This PluginBundle is like the @Default
with the following additions:

=over 4

=item *

Inserts a SUPPORT section to the POD just before AUTHOR

=item *

Adds the List Transformer

=item *

Enables WikiDoc formatting

=item *

Generates and collects stopwords

=back

It is roughly equivalent to:

	[WikiDoc]                 ; transform wikidoc sections to POD
	[@CorePrep]               ; [@Default]

	[Name]                    ; [@Default]
	[Version]                 ; [@Default]

	[Region  / prelude]       ; [@Default]

	[Generic / SYNOPSIS]      ; [@Default]
	[Generic / DESCRIPTION]   ; [@Default]
	[Generic / OVERVIEW]      ; [@Default]
	[Generic / USAGE]         ; Put USAGE section near the top

	[Collect / ATTRIBUTES]    ; [@Default]
	command = attr

	[Collect / METHODS]       ; [@Default]
	command = method

	[Collect / FUNCTIONS]     ; [@Default]
	command = func

	[Leftovers]               ; [@Default]

	[Region  / postlude]      ; [@Default]

	; custom section
	[Support]                 ; =head1 SUPPORT (bugs, cpants, git...)
	repository_content =
	repository_link = both

	[Authors]                 ; [@Default]
	[Legal]                   ; [@Default]

	[-Transformer]            ; enable =for :list
	transformer = List

	[-StopWords]              ; generate some stopwords and gather them together

=for Pod::Coverage mvp_bundle_config

=head1 AUTHOR

Randy Stauner <rwstauner@cpan.org>

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2010 by Randy Stauner.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut



Makefile.PL

use strict;
use warnings;

BEGIN { require 5.006; }

use ExtUtils::MakeMaker 6.31;



my %WriteMakefileArgs = (
  'ABSTRACT' => 'keep those pesky gophers out of your dists!',
  'AUTHOR' => 'Randy Stauner <rwstauner@cpan.org>',
  'BUILD_REQUIRES' => {
    'File::Find' => '0',
    'File::Temp' => '0',
    'ExtUtils::MakeMaker' => '6.31',
    'Test::More' => '0.96'
  },
  'CONFIGURE_REQUIRES' => {
    'ExtUtils::MakeMaker' => '6.31'
  },
  'DISTNAME' => 'Dist-Zilla-PluginBundle-GopherRepellent',
  'EXE_FILES' => [],
  'LICENSE' => 'perl',
  'NAME' => 'Dist::Zilla::PluginBundle::GopherRepellent',
  'PREREQ_PM' => {
    'Dist::Zilla' => '4.102345',
    'Dist::Zilla::Plugin::Authority' => '1.001',
    'Dist::Zilla::Plugin::Bugtracker' => '0',
    'Dist::Zilla::Plugin::CPANChangesTests' => '0',
    'Dist::Zilla::Plugin::CheckChangesHasContent' => '0.003',
    'Dist::Zilla::Plugin::CompileTests' => '1.100740',
    'Dist::Zilla::Plugin::DualBuilders' => '1.001',
    'Dist::Zilla::Plugin::Git::NextVersion' => '0',
    'Dist::Zilla::Plugin::GithubMeta' => '0.10',
    'Dist::Zilla::Plugin::KwaliteeTests' => '0',
    'Dist::Zilla::Plugin::MetaNoIndex' => '1.101130',
    'Dist::Zilla::Plugin::MetaProvides::Package' => '1.11044404',
    'Dist::Zilla::Plugin::MinimumPerl' => '0.02',
    'Dist::Zilla::Plugin::MinimumVersionTests' => '0',
    'Dist::Zilla::Plugin::NextRelease' => '0',
    'Dist::Zilla::Plugin::PkgVersion' => '0',
    'Dist::Zilla::Plugin::PodCoverageTests' => '0',
    'Dist::Zilla::Plugin::PodSpellingTests' => '0',
    'Dist::Zilla::Plugin::PodSyntaxTests' => '0',
    'Dist::Zilla::Plugin::PodWeaver' => '0',
    'Dist::Zilla::Plugin::PortabilityTests' => '0',
    'Dist::Zilla::Plugin::Prepender' => '1.100960',
    'Dist::Zilla::Plugin::ReportVersions::Tiny' => '1.01',
    'Dist::Zilla::Plugin::Repository' => '0.16',
    'Dist::Zilla::Plugin::TaskWeaver' => '0.101620',
    'Dist::Zilla::PluginBundle::Basic' => '0',
    'Dist::Zilla::Role::PluginBundle::Easy' => '0',
    'Dist::Zilla::Stash::PodWeaver' => '1.001000',
    'Moose' => '0',
    'Pod::Elemental' => '0.102360',
    'Pod::Elemental::Transformer::List' => '0',
    'Pod::Weaver' => '3.101632',
    'Pod::Weaver::Config::Assembler' => '0',
    'Pod::Weaver::Plugin::StopWords' => '1.001005',
    'Pod::Weaver::Plugin::Transformer' => '0',
    'Pod::Weaver::Plugin::WikiDoc' => '0',
    'Pod::Weaver::PluginBundle::Default' => '0',
    'Pod::Weaver::Section::Support' => '1.001'
  },
  'VERSION' => '1.002',
  'test' => {
    'TESTS' => 't/*.t'
  }
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.56) } ) {
  my $br = delete $WriteMakefileArgs{BUILD_REQUIRES};
  my $pp = $WriteMakefileArgs{PREREQ_PM};
  for my $mod ( keys %$br ) {
    if ( exists $pp->{$mod} ) {
      $pp->{$mod} = $br->{$mod} if $br->{$mod} > $pp->{$mod};
    }
    else {
      $pp->{$mod} = $br->{$mod};
    }
  }
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



